x11rb-protocol/src/utils.rs

```diff
@@ -11,27 +11,21 @@
 
 #[cfg(all(feature = "std", unix))]
 mod raw_fd_container {
-    use std::mem::forget;
-    use std::os::unix::io::{AsRawFd, IntoRawFd, RawFd};
+    use rustix::fd::OwnedFd;
+    use std::os::unix::io::{AsRawFd, RawFd};
 
     /// A simple wrapper around RawFd that closes the fd on drop.
     ///
     /// On non-unix systems, this type is empty and does not provide
     /// any method.
-    #[derive(Debug, Hash, PartialEq, Eq)]
-    pub struct RawFdContainer(RawFd);
-
-    impl Drop for RawFdContainer {
-        fn drop(&mut self) {
-            let _ = nix::unistd::close(self.0);
-        }
-    }
+    #[derive(Debug)]
+    pub struct RawFdContainer(OwnedFd);
```

Comment:
Do we actually still need `RawFdContainer` or should we switch to using `OwnedFd` directly? (I found out that rustix uses `io_lifetimes` which just re-exports the std type if the version is new enough, so this is basically already `OwnedFd` from `std`)

(And should that be done in a separate PR since it will be quite noise and thus "not the best thing" to entangle this with this one)

Reply:
I've switched to `OwnedFd` directly for now.